xcodeproj/internal/xcodeproj_incremental_rule.bzl

```diff
@@ -206,6 +206,26 @@ https://github.com/MobileNativeFoundation/rules_xcodeproj/issues/new?template=bu
 
 # Actions
 
+def _write_autogeneration_config_file(
+        actions,
+        config,
+        name):
+    content = ""
+    autogeneration_config_file = actions.declare_file(
+        "{}-autogeneration-config-file".format(name),
+    )
+    if "scheme_name_exclude_patterns" in config:
+        for pattern in config["scheme_name_exclude_patterns"]:
+            content += "{}\n".format(pattern)
+
+    content += "\n"
```

Comment:
We should instead use `args()` instead of building up a string.

Reply:
https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/3027/commits/69c66c5c4caa3d15d74766106ba1e4c9b4b7fc49

Reply:
Let me know if this is what you had in mind, used similar code paths in the repo as a reference.